src/Roslyn.Diagnostics.Analyzers/Core/DeclarePublicAPIAnalyzer.Impl.cs

```diff
@@ -361,15 +360,39 @@ private string GetSiblingNamesToRemoveFromUnshippedTextCore(ISymbol symbol)
             private string GetPublicApiName(ISymbol symbol)
             {
                 string publicApiName = symbol.ToDisplayString(s_publicApiFormat);
-
                 ITypeSymbol memberType = null;
-                if (symbol is IMethodSymbol)
+                if (symbol is IMethodSymbol method)
                 {
-                    memberType = ((IMethodSymbol)symbol).ReturnType;
+                    memberType = method.ReturnType;
+                    if (symbol.Language == LanguageNames.VisualBasic)
+                    {
+                       var theActualPropertySymbol = method.AssociatedSymbol;
+                       // Is the method a VB Setter Or Getter method?
+                        if (theActualPropertySymbol != null && theActualPropertySymbol is IPropertySymbol property)
+                        {
+                            // If so the use the property symbol instead, by doing so this make it
+                            // irrevalent if it an implemented property or auto-implemented property.
+                            // The entry in the PublicAPI file is the same.
+                            // Additionally it would also alert us to changes of readonly / writeonly.
+                            // eg;
+                            // Public Property [Property]() As Integer
+                            // to
+                            // Public ReadOnly Property [Property]() As Integer
+                            //
+                            // Note: Any property that is in the file prior to this change will report issue with the getter and setters.
```

Comment:
I don't like this breaking change - makes the public API files inconsistent between languages and introduces lot of unnecessary churn for all the analyzer consumers. We should think of alternate approaches.

Reply:
I am going to do run against roslyn so there is some info to gauge the impact.

Reply:
I don't think it matters: as a team we're pretty resistant to format changes, and I think it should continue to apply here. Doing a format change will just cause headaches, especially when we have different branches trying to merge around.